thorlcr/graph/thgraphslave.cpp

```diff
@@ -140,10 +140,29 @@ void CSlaveActivity::setInput(unsigned index, CActivityBase *inputActivity, unsi
     else
         outLink.set(((CSlaveActivity *)inputActivity)->queryOutput(inputOutIdx));
     assertex(outLink);
+
     while (inputs.ordinality()<=index) inputs.append(NULL);
+
     inputs.replace(outLink.getClear(), index);
 }
 
+void CSlaveActivity::appendOutput(IThorDataLink *itdl)
+{
+    if (queryJob().getOptBool("TRACEROWS"))
+    {
+        const unsigned numTraceRows = queryJob().getOptInt("numTraceRows", 10);
+        outputs.append(new CTracingThorDataLink(itdl, queryHelper(), numTraceRows));
```

Comment:
Will this work if an output ever gets cast (or query interfaced) back to its original base class?

What I am seeing is that the metaInfo (counts and recordSize etc.) is not getting reported within the graphs and suspect that something casting *CTracingThorDataLink to *CThorDataLink (via queryOutput?) and if successful pulls the metaInfo from it?